gno.land/cmd/gnoland/root.go

```diff
@@ -1,42 +1,57 @@
 package main
 
 import (
-	"context"
-	"fmt"
+	//	"context"
+
+	"flag"
 	"os"
 
+	tmcfg "github.com/gnolang/gno/tm2/pkg/bft/config"
 	"github.com/gnolang/gno/tm2/pkg/commands"
-	"github.com/peterbourgon/ff/v3"
-	"github.com/peterbourgon/ff/v3/fftoml"
+	"github.com/gnolang/gno/tm2/pkg/log"
 )
 
-func main() {
-	io := commands.NewDefaultIO()
```

Comment:
To retain the benefits of commands.IO while aligning with your approach, I propose introducing a commands.OutErr that disallows stdin setting and enables compiler verification.

If you're on board, we can proceed with my initial work to transition IO from a struct to an interface, detailed in this pull request: https://github.com/gnolang/gno/pull/1270.

Additionally, we can create a variant that functions as a "read-only IO," (commands.OutErr or better name) analogous to Golang's IO Writer, Reader, and ReadWriter, for commands that distinguish between 'out' and 'err.'

Reply:
Do you intend `commands.OutErr` is an interface, subset of what would be `commands.IO`, which does not have `Stdin`?